src/prpy/planning/tsr.py

```diff
@@ -141,41 +141,46 @@ def PlanToTSR(self, robot, tsrchains, tsr_timeout=2.0,
         valid_solutions = ik_solutions[valid_idxs, :]
         ranked_indices = numpy.argsort(valid_scores)
         ranked_ik_solutions = valid_solutions[ranked_indices, :]
-
-        # Group the IK solutions into sets of the specified size
-        # (plan for each set of IK solutions together).
-        ranked_ik_solution_sets = [
-            ranked_ik_solutions[i:i + chunk_size, :]
-            for i in range(0, ranked_ik_solutions.shape[0], chunk_size)
-        ]
-
-        num_attempts = min(len(ranked_ik_solution_sets), num_attempts)
-        ik_set_list = enumerate(ranked_ik_solution_sets[:num_attempts])
+        ranked_ik_solutions = list(ranked_ik_solutions)
 
         # Configure the robot to use the active manipulator for planning.
+        from openravepy import CollisionOptions, CollisionOptionsStateSaver
+
         p = openravepy.KinBody.SaveParameters
-        with robot.CreateRobotStateSaver(p.ActiveDOF):
+        with robot.CreateRobotStateSaver(p.ActiveDOF),\
+              CollisionOptionsStateSaver(self.env.GetCollisionChecker(), 
+                                         CollisionOptions.ActiveDOFs):
+            
             robot.SetActiveDOFs(manipulator.GetArmIndices())
 
-            # Try planning to each solution set in descending cost order.
-            for i, ik_set in ik_set_list:
+            for i_attempt in xrange(num_attempts):
+                ik_set = []
+                while len(ik_set) < chunk_size and ranked_ik_solutions:
```

Comment:
Can we add a comment explaining that this loop tries to accumulate up to the next `chunk_size` of `ranked_ik_solutions` that are not in collision?